encrypt.go

```diff
@@ -163,8 +163,8 @@ func (es *encryptStream) checkReceivers(v []BoxPublicKey) error {
 		tot++
 	}
 
-	// Don't allow more than 2^31 receivers.
-	if tot >= 0x7fffffff {
+	// Don't allow more than 2^32-1 receivers.
+	if tot > ((1 << 32) - 1) {
```

Comment:
Is this correct on 32-bit systems?

Reply:
Yeap, https://golang.org/ref/spec#Constants says constants are calculated with at least 256-bit precision, and we only compare it to a `uint64`. Tested with `GOARCH=i386` to make sure.